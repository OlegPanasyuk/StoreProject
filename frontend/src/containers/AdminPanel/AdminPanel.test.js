import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';

import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import AdminPanelConnected, { AdminPanel } from './AdminPanel';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    match: {
        path: ''
    },
    errors: [],
    user: {
        token: ''
    }
}) {
    const component = shallow(<AdminPanel {...props} />);
    return { component, props };
}

describe('AdminPanel --- snapshots', () => {
    it('Component must be rendered', () => {
        const obj = setupComponent();
        expect(AdminPanel.propTypes.user).toBe(PropTypes.object);
        expect(AdminPanel.propTypes.errors).toBe(PropTypes.array);
        expect(AdminPanel.propTypes.match).toBe(PropTypes.object);
        expect(shallowToJson(obj.component)).toMatchSnapshot();
    });
});

describe('AdminPanelConnected', () => {
    const initialState = {
        errorReducers: {
            Errors: []
        },
        adminPanel_User: {
            message: 'Auth Passed',
            token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImFkbWluQGFkbWluLmNvbSIsInJvbGUiOiJTdXBlckFkbWluIiwiaWF0IjoxNTU4MzQxNDIxLCJleHAiOjE1NTgzNDE1NDF9.JRn4XDwiwNonZCxU65cjJ1NgZ5zEnJ35KSgCpxnLwMU',
            role: 'SuperAdmin',
            username: 'Admin'
        }
    };

    const mockStore = configureStore();
    let store; let
        wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(
            <Provider store={store}>
                <Router>
                    <AdminPanelConnected match={{
                        path: ''
                    }}
                    />
                </Router>
            </Provider>
        );
    });
    it('+++ render the connected(SMART) component', () => {
        expect(wrapper.find(AdminPanelConnected).length).toEqual(1);
    });
});
