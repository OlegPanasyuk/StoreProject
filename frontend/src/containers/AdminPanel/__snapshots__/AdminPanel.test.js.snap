// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPanel --- snapshots Component must be rendered 1`] = `
<BrowserRouter>
  <Container
    as="div"
    fluid={false}
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(LoginForm)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        path="/login"
      />
      <Route
        render={[Function]}
      />
    </Switch>
  </Container>
</BrowserRouter>
`;
